detection_rules/version_lock.py

```diff
@@ -224,65 +219,73 @@ def add_changes(r, *msg):
+                        "sha256": lock_from_file["sha256"],
+                        "version": lock_from_file["version"],
+                        "type": lock_from_file["type"]
                     }
-                    existing_rule_lock.setdefault("previous", {})
+                    lock_from_file.setdefault("previous", {})
 
                     # move the current locked info into the previous section
-                    existing_rule_lock["previous"][str(latest_locked_stack_version)] = previous_lock_info
+                    lock_from_file["previous"][str(latest_locked_stack_version)] = previous_lock_info
 
                     # overwrite the "latest" part of the lock at the top level
                     # TODO: would need to preserve space here as well if supporting forked version spacing
-                    existing_rule_lock.update(current_rule_lock, min_stack_version=str(min_stack))
-                    add_changes(
-                        rule,
+                    lock_from_file.update(lock_from_rule, min_stack_version=str(min_stack))
+                    new_version = lock_from_rule['version']
+                    log_changes(
+                        rule, route, new_version,
                         f'previous {latest_locked_stack_version} saved as version: {previous_lock_info["version"]}',
                         f'current min_stack updated to {min_stack}'
                     )
 
                 elif min_stack < latest_locked_stack_version:
                     route = 'C'
                     # 4) on an old stack, after a breaking change has been made (updated fork)
-                    assert str(min_stack) in existing_rule_lock.get("previous", {}), \
+                    assert str(min_stack) in lock_from_file.get("previous", {}), \
                         f"Expected {rule.id} @ v{min_stack} in the rule lock"
 
                     # TODO: Figure out whether we support locking old versions and if we want to
                     #       "leave room" by skipping versions when breaking changes are made.
                     #       We can still inspect the version lock manually after locks are made,
                     #       since it's a good summary of everything that happens
-                    existing_rule_lock["previous"][str(min_stack)] = current_rule_lock
-                    existing_rule_lock.update(current_rule_lock)
-                    add_changes(rule, f'previous version {min_stack} updated version to {current_rule_lock["version"]}')
+
+                    # if version bump collides with future bump, fail
+                    # if space, change and log
+                    info_from_rule = (lock_from_rule['sha256'], lock_from_rule['version'])
+                    info_from_file = (lock_from_file["previous"][str(min_stack)]['sha256'],
+                                      lock_from_file["previous"][str(min_stack)]['version'])
+                    if info_from_rule != info_from_file:
```

Comment:
previously, this would _always_ overwrite with the current lock info. This adds an additional check to verify the actual change before writing (should effectively be a NOP if everything is in order)